lib/facter/root_home.rb

```diff
@@ -1,49 +1,13 @@
 # frozen_string_literal: true
 
-# root_home.rb
-module Facter::Util::RootHome
-  # @summary
-  #   A facter fact to determine the root home directory.
-  #   This varies on PE supported platforms and may be
-  #   reconfigured by the end user.
-  class << self
-    # determines the root home directory
-    def returnt_root_home
-      root_ent = Facter::Core::Execution.execute('getent passwd root')
-      # The home directory is the sixth element in the passwd entry
-      # If the platform doesn't have getent, root_ent will be nil and we should
-      # return it straight away.
-      root_ent && root_ent.split(':')[5]
-    end
-  end
-end
-
-Facter.add(:root_home) do
-  setcode { Facter::Util::RootHome.returnt_root_home }
-end
-
-Facter.add(:root_home) do
-  confine kernel: :darwin
-  setcode do
-    str = Facter::Core::Execution.execute('dscacheutil -q user -a name root')
-    hash = {}
-    str.split("\n").each do |pair|
-      key, value = pair.split(%r{:})
-      hash[key] = value
-    end
-    hash['dir'].strip
-  end
-end
-
 Facter.add(:root_home) do
```

Comment:
Nice removal of code @ekohl.

What do you think about confining the fact to osx and linux? I noticed the test just checks for `nil` when `:kernel` is Windows.

Reply:
Today there's also fact for AIX and I'd expect this code to work on any unix (like Solaris). However, I can't test that since I don't have access to those systems. Any you could confine it to non-Windows but I think this is effectively the same.

Reply:
Fair point!